aziotctl/src/internal/check/checks/supported_platform.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+use anyhow::{anyhow, Result};
+use serde::Serialize;
+
+use crate::internal::check::{CheckResult, Checker, CheckerCache, CheckerMeta, CheckerShared};
+
+#[derive(Serialize, Default)]
+pub struct SupportedPlatform {}
+
+#[async_trait::async_trait]
+impl Checker for SupportedPlatform {
+    fn meta(&self) -> CheckerMeta {
+        CheckerMeta {
+            id: "supported-platform",
+            description: "production readiness: running on supported platform",
+        }
+    }
+
+    async fn execute(&mut self, shared: &CheckerShared, cache: &mut CheckerCache) -> CheckResult {
+        self.inner_execute(shared, cache)
+            .await
+            .unwrap_or_else(CheckResult::Failed)
+    }
+}
+
+impl SupportedPlatform {
+    async fn inner_execute(
+        &mut self,
+        _shared: &CheckerShared,
+        _cache: &mut CheckerCache,
+    ) -> Result<CheckResult> {
+        use os_info::{Type, Version};
+
+        enum Tier {
+            Tier1,
+            Tier2,
+            Tier3,
+        }
+
+        let info = os_info::get();
+        let tier = match info.os_type() {
+            Type::CentOS | Type::Debian | Type::Ubuntu => {
+                if match info.os_type() {
+                    Type::CentOS => {
+                        matches!(info.version(), Version::Custom(v) if v == "7")
+                    }
+                    Type::Debian => {
+                        matches!(info.version(), Version::Custom(v) if v == "9" || v == "10")
+                    }
+                    Type::Ubuntu => {
+                        matches!(info.version(), Version::Custom(v) if v == "20.04" || v == "18.04")
+                    }
+                    _ => unreachable!(),
+                } {
+                    Tier::Tier1
+                } else {
+                    Tier::Tier2
+                }
+            }
+            Type::Alpine
+            | Type::Amazon
+            | Type::Arch
+            | Type::Fedora
+            | Type::Linux
+            | Type::Manjaro
+            | Type::Mint
+            | Type::openSUSE
+            | Type::OracleLinux
+            | Type::Pop
+            | Type::Redhat
+            | Type::RedHatEnterprise
+            | Type::SUSE => Tier::Tier2,
+            _ => Tier::Tier3,
+        };
+
+        Ok(match tier {
+            Tier::Tier1 => CheckResult::Ok,
+            Tier::Tier2 => CheckResult::Warning(anyhow!(
+                "Running on a partially supported, Tier 2 platform ({}). lorem ipsum",
```

Comment:
Can we add the URL to the tiering docs?

Reply:
https://docs.microsoft.com/en-us/azure/iot-edge/support?view=iotedge-2018-06

Reply:
You should check with Kelly if that is the URL to use or if there a lasting one that can be redirected as needed

Reply:
We need an aziot equivalent of https://aka.ms/iotedge-platsup . I'll make https://aka.ms/aziot-platsup for now and we can redirect it to aziot-specific documentation later (same approach we used for https://github.com/Azure/iot-identity-service/pull/179#discussion_r600971063 )

Reply:
Done. @daprilik Use https://aka.ms/aziot-platsup

Reply:
Awesome. Didn't know we already had this list. I'll update the PR tomorrow.